app/views/locations/index.html.erb

```diff
@@ -50,14 +50,20 @@
         <% @locations.each do |location| %>
           <%= table.body do |body| %>
             <%= body.row do |row| %>
-              <% row.cell(text: simple_format(location_cell_postcode(location, scheme_location_path(@scheme, location)), { class: "govuk-!-font-weight-bold" }, wrapper_tag: "div")) %>
+              <% row.cell(text: simple_format(location_cell_postcode(location, if location.confirmed
+                                                                                 scheme_location_path(@scheme, location)
+                                                                               else
+                                                                                 location.postcode.present? ? scheme_location_check_answers_path(@scheme, location, route: "locations") : scheme_location_postcode_path(@scheme, location)
+                                                                               end), { class: "govuk-!-font-weight-bold" }, wrapper_tag: "div")) %>
               <% row.cell(text: location.id) %>
               <% row.cell(text: status_tag(location.status)) %>
               <% end %>
           <% end %>
         <% end %>
       <% end %>
-      <%= govuk_button_link_to "Add a location", new_scheme_location_path(@scheme), secondary: true %>
+      <%= form_with model: @scheme, url: scheme_locations_path(@scheme), method: "post", local: true do |f| %>
+        <%= f.govuk_submit "Add a location", secondary: true %>
```

Comment:
This can be a link_to with `method: :post` and Rails will largely generate what you've written here by itself.

https://apidock.com/rails/ActionView/Helpers/UrlHelper/link_to